server/src/test/java/io/spine/system/server/CommandMemoizingTap.java

```diff
@@ -18,27 +18,42 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-package io.spine.server.commandbus;
+package io.spine.system.server;
 
 import com.google.protobuf.Message;
-import io.spine.core.Command;
-import io.spine.testing.client.TestActorRequestFactory;
-import org.junit.jupiter.api.DisplayName;
+import io.spine.core.Ack;
+import io.spine.core.CommandEnvelope;
+import io.spine.core.CommandId;
+import io.spine.server.bus.BusFilter;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.collect.Maps.newHashMap;
 
 /**
- * @author Alexander Yevsyukov
+ * A {@link BusFilter} which remembers all the accepted command messages and never halts the command
+ * processing.
+ *
+ * @author Dmytro Dashenkov
  */
-@DisplayName("Single tenant CommandStore should")
-class SingleTenantCommandStoreTest extends CommandStoreTest {
+final class CommandMemoizingTap implements BusFilter<CommandEnvelope> {
+
+    private final Map<CommandId, Message> commandMessages = newHashMap();
 
-    SingleTenantCommandStoreTest() {
-        super(false);
+    /**
+     * Looks up the command message by the command ID.
+     */
+    <M extends Message> Optional<M> find(CommandId commandId) {
+        Message commandMessage = commandMessages.get(commandId);
+        @SuppressWarnings("unchecked")
```

Comment:
Please explain how we know this suppression is safe.

I know the ErrorProne complaints in the cases like `<M extends Message> Optional<M>  doSomething(..)`